tests/helpers/test_script.py

```diff
@@ -4752,7 +4755,7 @@ async def test_shutdown_after(
     expected_trace = {
         "0": [{"result": {"delay": 120.0, "done": False}}],
     }
-    assert_action_trace(expected_trace)
+    assert_action_trace(expected_trace, "cancelled")
```

Comment:
The `await script_obj.async_stop()` called from the stop event works as expected now and the state now correctly reflects cancelled